# Explore time series contiguity of mini dataset

In [1]:
import os
import iris
import xarray as xr

In [2]:
filepath = '/data/cssp-china/mini-dataset-24-01-19/20CR/daily'
PRECIP_STASH = 'm01s05i216'

In [4]:
files = sorted(os.listdir(filepath))

In [7]:
for file in [files[0], files[1], files[-2], files[-1]]:
    cube, = iris.load(os.path.join(filepath, file), iris.AttributeConstraint(STASH=PRECIP_STASH))
    
    display([c for c in cube.coord('time').cells()])
    print()

[Cell(point=real_datetime(1851, 1, 1, 12, 0), bound=(real_datetime(1851, 1, 1, 0, 0), real_datetime(1851, 1, 2, 0, 0))),
 Cell(point=real_datetime(1851, 1, 2, 12, 0), bound=(real_datetime(1851, 1, 2, 0, 0), real_datetime(1851, 1, 3, 0, 0))),
 Cell(point=real_datetime(1851, 1, 3, 12, 0), bound=(real_datetime(1851, 1, 3, 0, 0), real_datetime(1851, 1, 4, 0, 0))),
 Cell(point=real_datetime(1851, 1, 4, 12, 0), bound=(real_datetime(1851, 1, 4, 0, 0), real_datetime(1851, 1, 5, 0, 0)))]

[Cell(point=real_datetime(1851, 1, 5, 12, 0), bound=(real_datetime(1851, 1, 5, 0, 0), real_datetime(1851, 1, 6, 0, 0))),
 Cell(point=real_datetime(1851, 1, 6, 12, 0), bound=(real_datetime(1851, 1, 6, 0, 0), real_datetime(1851, 1, 7, 0, 0))),
 Cell(point=real_datetime(1851, 1, 7, 12, 0), bound=(real_datetime(1851, 1, 7, 0, 0), real_datetime(1851, 1, 8, 0, 0))),
 Cell(point=real_datetime(1851, 1, 8, 12, 0), bound=(real_datetime(1851, 1, 8, 0, 0), real_datetime(1851, 1, 9, 0, 0))),
 Cell(point=real_datetime(1851, 1, 9, 12, 0), bound=(real_datetime(1851, 1, 9, 0, 0), real_datetime(1851, 1, 10, 0, 0))),
 Cell(point=real_datetime(1851, 1, 10, 12, 0), bound=(real_datetime(1851, 1, 10, 0, 0), real_datetime(1851, 1, 11, 0, 0))),
 Cell(point=real_datetime(1851, 1, 11, 12, 0), bound=(real_datetime(1851, 1, 11, 0, 0), real_datetime(1851, 1, 12, 0, 0))),
 Cell(point=real_datetime(1851, 1, 12, 12, 0), bound=(real_datetime(1851, 1, 12, 0, 0), real_datetime(1851, 1, 13, 0, 0))),
 Cell(point=real_datet

[Cell(point=real_datetime(2010, 12, 16, 12, 0), bound=(real_datetime(2010, 12, 16, 0, 0), real_datetime(2010, 12, 17, 0, 0))),
 Cell(point=real_datetime(2010, 12, 17, 12, 0), bound=(real_datetime(2010, 12, 17, 0, 0), real_datetime(2010, 12, 18, 0, 0))),
 Cell(point=real_datetime(2010, 12, 18, 12, 0), bound=(real_datetime(2010, 12, 18, 0, 0), real_datetime(2010, 12, 19, 0, 0))),
 Cell(point=real_datetime(2010, 12, 19, 12, 0), bound=(real_datetime(2010, 12, 19, 0, 0), real_datetime(2010, 12, 20, 0, 0))),
 Cell(point=real_datetime(2010, 12, 20, 12, 0), bound=(real_datetime(2010, 12, 20, 0, 0), real_datetime(2010, 12, 21, 0, 0))),
 Cell(point=real_datetime(2010, 12, 21, 12, 0), bound=(real_datetime(2010, 12, 21, 0, 0), real_datetime(2010, 12, 22, 0, 0))),
 Cell(point=real_datetime(2010, 12, 22, 12, 0), bound=(real_datetime(2010, 12, 22, 0, 0), real_datetime(2010, 12, 23, 0, 0))),
 Cell(point=real_datetime(2010, 12, 23, 12, 0), bound=(real_datetime(2010, 12, 23, 0, 0), real_datetime(2010, 1

[Cell(point=real_datetime(2010, 12, 26, 12, 0), bound=(real_datetime(2010, 12, 26, 0, 0), real_datetime(2010, 12, 27, 0, 0))),
 Cell(point=real_datetime(2010, 12, 27, 12, 0), bound=(real_datetime(2010, 12, 27, 0, 0), real_datetime(2010, 12, 28, 0, 0))),
 Cell(point=real_datetime(2010, 12, 28, 12, 0), bound=(real_datetime(2010, 12, 28, 0, 0), real_datetime(2010, 12, 29, 0, 0))),
 Cell(point=real_datetime(2010, 12, 29, 12, 0), bound=(real_datetime(2010, 12, 29, 0, 0), real_datetime(2010, 12, 30, 0, 0))),
 Cell(point=real_datetime(2010, 12, 30, 12, 0), bound=(real_datetime(2010, 12, 30, 0, 0), real_datetime(2010, 12, 31, 0, 0))),
 Cell(point=real_datetime(2010, 12, 31, 12, 0), bound=(real_datetime(2010, 12, 31, 0, 0), real_datetime(2011, 1, 1, 0, 0)))]

## So, time is contiguous (for daily data)
But the block sizes at the start and end of the series are not 10 days in length, which makes an issue for starting the zarray (unless I drop first 4 days of 1851)